src/iqm/iqm_client/models.py

```diff
@@ -529,6 +517,92 @@ class QuantumArchitecture(BaseModel):
+
+    Note: in this mode, the number of shots returned after execution will be less or equal to the requested amount
+    due to the post-selection based on heralding data."""
+
+
+class MoveGateValidationMode(Enum):
+    """MOVE gate validation mode for circuit compilation. This options is meant for advanced users."""
+
+    STRICT: Final[str] = 'strict'
+    """Perform standard MOVE gate validation: MOVE gates must only appear in sandwiches, with no gates acting on the
+    MOVE qubit inside the sandwich."""
+    ALLOW_PRX: Final[str] = 'allow_prx'
+    """Allow PRX gates on the MOVE qubit inside MOVE sandwiches during validation."""
+    NONE: Final[str] = 'none'
+    """Do not perform any MOVE gate validation."""
+
+
+class MoveGateFrameTrackingMode(Enum):
+    """MOVE gate frame tracking mode for circuit compilation. This option is meant for advanced users."""
+
+    FULL: Final[str] = 'full'
+    """Perform complete MOVE gate frame tracking."""
+    NO_DETUNING_CORRECTION: Final[str] = 'no_detuning_correction'
+    """Do not add the phase detuning corrections to the pulse schedule for the MOVE gate. The user is expected to do
+    these manually."""
+    NONE: Final[str] = 'none'
+    """Do not perform any MOVE gate frame tracking. The user is expected to do these manually."""
+
+
+@dataclass(frozen=True)
+class CircuitCompilationOptions:
+    """Various discrete options for quantum circuit compilation to pulse schedule.
+
+    max_circuit_duration_over_t2:  Circuits are disqualified on the server if they are longer than this ratio
+        of the T2 time of the qubits. Setting this value to ``0.0`` turns off circuit duration checking.
+        The default value ``None`` instructs server to use server's default value in the checking.
+    heralding_mode: Heralding mode to use during the execution.
+    move_gate_validation: The MOVE gate validation mode for circuit compilation.
+    move_gate_frame_tracking: The MOVE gate frame tracking mode for circuit compilation.
+    """
+
+    max_circuit_duration_over_t2: Optional[float] = None
+    heralding_mode: HeraldingMode = HeraldingMode.NONE
+    move_gate_validation: MoveGateValidationMode = MoveGateValidationMode.STRICT
+    move_gate_frame_tracking: MoveGateFrameTrackingMode = MoveGateFrameTrackingMode.FULL
+
+    def __post_init__(self):
+        """Validate the options."""
+        if self.move_gate_frame_tracking == MoveGateFrameTrackingMode.FULL and self.move_gate_validation not in [
+            MoveGateValidationMode.STRICT,
+            MoveGateValidationMode.ALLOW_PRX,
+            None,
+        ]:
+            raise ValueError(
+                'Unable to perform full MOVE gate frame tracking if MOVE gate validation is not'
+                + ' "strict" or "allow_prx".'
+            )
+
+    def _validate_sensible_use(self, arch: QuantumArchitectureSpecification, circuit: 'CircuitBatch') -> None:
```

Comment:
```suggestion
    def _validate_sensible_use(self, arch: QuantumArchitectureSpecification, circuit: CircuitBatch) -> None:
```
Annotations do not need to be strings if you use `from __future__ import annotations` in the beginning of the module.

Reply:
Also elsewhere.